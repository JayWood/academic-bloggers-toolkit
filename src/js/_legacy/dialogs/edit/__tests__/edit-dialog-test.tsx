import { shallow, ShallowWrapper } from 'enzyme';
import toJSON, { OutputMapper } from 'enzyme-to-json';
import React from 'react';
import EditDialog from '..';

const defaultData: CSL.Data = {
    id: '12345',
    title: 'Test Title',
    type: 'article-journal',
};

const setup = (data = defaultData) => {
    const onSubmit = jest.fn();
    const component = shallow(<EditDialog data={data} onSubmit={onSubmit} />);
    return {
        component,
        instance: component.instance() as EditDialog,
        onSubmit,
    };
};

const mapper: OutputMapper = json => {
    if (json.props.meta) {
        delete json.props.meta;
    }
    return json;
};

const J = (component: ShallowWrapper) =>
    toJSON(component, { noKey: true, map: mapper });

describe('<EditDialog />', () => {
    const BASELINE = J(setup().component);
    it('should match baseline snapshot', () => {
        expect(BASELINE).toMatchSnapshot();
    });
    it('should submit', () => {
        const { component, onSubmit } = setup();
        const form = component.find('form');
        const preventDefault = jest.fn();
        form.simulate('submit', { preventDefault });
        expect(preventDefault).toHaveBeenCalled();
        expect(onSubmit).toHaveBeenCalledWith(defaultData);
    });
});
