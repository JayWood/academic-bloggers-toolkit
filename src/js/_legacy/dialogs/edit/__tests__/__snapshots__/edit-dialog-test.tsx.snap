// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditDialog /> should match baseline snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <ContributorList
      citationType="article-journal"
      people={Array []}
    />
    <MetaFields />
  </div>
  <ActionBar
    align="right"
  >
    <Button
      flat={true}
      label="Confirm"
      primary={true}
      type="submit"
    >
      Confirm
    </Button>
  </ActionBar>
</form>
`;
